src/components/sidebar/Header.js

```diff
@@ -18,6 +21,11 @@ export default function Header({
   handleDrawerClose,
   drawerOpen,
 }) {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(setUserEmail(user && user.attributes.email));
```

Comment:
The header is supposed to be an individual component. With this code, the whole app depends on its existence. Is this okay?

Reply:
I find that it's unnecessary to use useEffect here since the only time that user would be changed is when user log out and log back in, and the whole app would re-render again at that time. would removing the useEffect does the trick?